packages/tree/src/model/node.ts

```diff
@@ -0,0 +1,514 @@
+      this.childNodes.push(child as Node)
+    } else {
+      this.childNodes.splice(index, 0, child as Node)
+    }
+
+    this.updateLeafState()
+  }
+
+  insertBefore(child: FakeNode | Node, ref: Node): void {
+    let index
+    if (ref) {
+      index = this.childNodes.indexOf(ref)
+    }
+    this.insertChild(child, index)
+  }
+
+  insertAfter(child: FakeNode | Node, ref: Node): void {
+    let index
+    if (ref) {
+      index = this.childNodes.indexOf(ref)
+      if (index !== -1) index += 1
+    }
+    this.insertChild(child, index)
+  }
+
+  removeChild(child: Node): void {
+    const children = this.getChildren() || []
+    const dataIndex = children.indexOf(child.data)
+    if (dataIndex > -1) {
+      children.splice(dataIndex, 1)
+    }
+
+    const index = this.childNodes.indexOf(child)
+
+    if (index > -1) {
+      this.store && this.store.deregisterNode(child)
+      child.parent = null
+      this.childNodes.splice(index, 1)
+    }
+
+    this.updateLeafState()
+  }
+
+  removeChildByData(data: TreeNodeData): void {
+    let targetNode: Node = null
+
+    for (let i = 0; i < this.childNodes.length; i++) {
+      if (this.childNodes[i].data === data) {
+        targetNode = this.childNodes[i]
+        break
+      }
+    }
+
+    if (targetNode) {
+      this.removeChild(targetNode)
+    }
+  }
+
+  expand(callback?: () => void, expandParent?: boolean): void {
+    const done = (): void => {
+      if (expandParent) {
+        let parent = this.parent
+        while (parent.level > 0) {
+          parent.expanded = true
+          parent = parent.parent
+        }
+      }
+      this.expanded = true
+      if (callback) callback()
+    }
+
+    if (this.shouldLoadData()) {
+      this.loadData(data => {
+        if (data instanceof Array) {
+          if (this.checked) {
+            this.setChecked(true, true)
+          } else if (!this.store.checkStrictly) {
+            reInitChecked(this)
+          }
+          done()
+        }
+      })
+    } else {
+      done()
+    }
+  }
+
+  doCreateChildren(array: TreeNodeData[], defaultProps: TreeNodeLoadedDefaultProps = {}): void {
+    array.forEach(item => {
+      this.insertChild(objectAssign({ data: item }, defaultProps), undefined, true)
+    })
+  }
+
+  collapse(): void {
+    this.expanded = false
+  }
+
+  shouldLoadData(): boolean {
+    return this.store.lazy === true && this.store.load && !this.loaded
+  }
+
+  updateLeafState(): void {
+    if (this.store.lazy === true && this.loaded !== true && typeof this.isLeafByUser !== 'undefined') {
+      this.isLeaf = this.isLeafByUser
+      return
+    }
+    const childNodes = this.childNodes
+    if (!this.store.lazy || (this.store.lazy === true && this.loaded === true)) {
+      this.isLeaf = !childNodes || childNodes.length === 0
+      return
+    }
+    this.isLeaf = false
+  }
+
+  setChecked(value?: boolean | string, deep?: boolean, recursion?: boolean, passValue?: boolean) {
```

Comment:
The calling site of `setChecked` has several boolean arguments without description. Could we make it better?